### DISCUSSION TITLE
webglstudio interrogation for G3Develop

### DISCUSSION THREAD

**Discussion by lv2lrn4lf** [4ian/GDevelop#3948] [Archive]
webglstudio interrogation for G3Develop

Came across webglstudio, it's a gui editor to make projects of animations, applications, and even games using a gui, webgl, glsl and javascript to make what you want. Would like to see some of these feature interrogated in gdevelop.. g3develop maybe.

G3Develop having 3D capability. Also, the performance this software display is amazing and gdevelop could benefit from using even one of it's modules rather that the shader or graph part of the software.

I'm guessing the opposition might be this software and gdevelop are fundamentally different and trying to integrate this or any part of this into the engine will push us to have to change the engine and interm gdevelop as a whole.

Despite the opposition, I think gdevelop would benefit from having some of these components like the shaders graphic for more power over image manipulation in gdevelop.... Either way it's pretty cool.

**Comment by Silver-Streak** [comment]
While webglstudio is a fantastic project, you are slightly missing the mark on what you think the opposition would be. 

Webglstudio isnt "another tool to integrate into GDevelop".

Webglstudio is a basically completely different game engine. It is using litescene.js as the renderer. Meaning that you are asking for another game engine entirely to be integrated into GDevelop (which isn't how things really work)

There *may* be something to discuss about integrating litescene, except that would mean replacing the renderer entirely, or figuring out how to stitch PixiJS and Litescene together (as litescene is 3D focused and has limited 2D camera support). 

As far as I can see, the problems with this would be quite numerous:
1. Replacing the renderer would require adapting all functions, logic code generation, and internal extensions/plugins to the new renderer.
2. Creating some form of Pixi/Litescene hybrid would require creating and maintaining an entirely new branch of both. Based off prior discussions from the GDevelop team, they generally want to avoid making custom branches of existing libraries. Otherwise they get more and more overhead to do even simple updates, or their branches become more and more stale from the current versions.
3. Perhaps the largest hurdle for this is that there are currently no plans nor desire to change GDevelop into a 3D or hybrid engine. The GDevelops team current states goal is to make it one of the best 2D engines they can. 


If number 3 were to ever change, I'd expect the team would want to use something directly compatible with PixiJS that doesn't need a replacement or custom integration, like PIXI3D. But that is a *huge* if. They will probably want to be more feature complete on 2D before anything else is entertained.

**Reply by lv2lrn4lf** [reply]
understandable... in case no one here was aware of this amazing tool... I call it amazing cause it allow me to do the same thing gdevelop do... write in javascript and program within the browser. 

I come to see the biggest strength is what the developers want you to do and that is make a game using no code at all. gdevelop is amazing and simple and I can see it can house bigger games/project if you watch the performance carefully. WebglStudio is simple but lack learning resource. gDevelop have learning resource to get you up and running. If any part of webglstudio can be use to better gdevelop, just in case that was possible, wanted to bring this to the developers attentions in case they didn't know... or who ever else this might help. 

But as you stated, and seeing you have knowledge of this tool already, than the over head would be to much of a distractions on advancing features than putting another wheel in it.
